js_modules/dagster-ui/packages/ui-core/src/gantt/GanttChart.tsx

```diff
@@ -289,17 +291,24 @@ const GanttChartInner = (props: GanttChartInnerProps) => {
   // The `layout` we receive has been laid out and the rows / "waterfall" are final,
   // but it doesn't incorporate the display scale or run metadata. We stretch and
   // shift the layout boxes using this data to create the final layout for display.
-  const layout = adjustLayoutWithRunMetadata(
-    props.layout,
-    options,
-    metadata || EMPTY_RUN_METADATA,
-    scale,
-    nowMs,
+  const layout = useThrottledMemo(
```

Comment:
This looks good to me, but I'm scared that this creates scenarios where a box can exist in props.layout and it does NOT exist in the adjusted layout for several renders (until this throttle fires).   I don't know offhand that that breaks anything, but it's not really a case this code was tested for.

Before this merges, it'd be nice to write a bit of code that takes one of the boxes in the returned `layout` and deletes it, just to see what happens to the remainder of the rendering downstream. 

I think this use case may happen in the wild with dynamic fan-out of ops where we create the boxes as the logs appear that reference them.

Reply:
>This looks good to me, but I'm scared that this creates scenarios where a box can exist in props.layout and it does NOT exist in the adjusted layout for several renders (until this throttle fires). I don't know offhand that that breaks anything, but it's not really a case this code was tested for.

I read through the code a few times while adding this and there shouldn't be any issue here because `props.layout` isn't passed to any other part of the UI so there's no chance for there to be a parallel and inconsistent data structure.